data/src/main/java/com/linkedin/data/schema/validation/ValidateDataAgainstSchema.java

```diff
@@ -278,13 +292,82 @@ protected void recurseRecord(DataElement element, RecordDataSchema schema, DataM
       {
         String key = entry.getKey();
         RecordDataSchema.Field field = schema.getField(key);
+        Object value = entry.getValue();
         if (field != null)
         {
-          Object value = entry.getValue();
           DataSchema childSchema = field.getType();
           childElement.setValueNameSchema(value, key, childSchema);
           validate(childElement, childSchema, value);
         }
+        else
```

Comment:
Are you sure that this is the only area (recurseRecord()) where you have to add this?